### PR TITLE
Fix slow `ilab` CLI startup

### PR DESCRIPTION
**Pull request by tiran**
Avoid importing `llama_cpp` by importing the `ServerException` from its
original module.

Defer import of `run_training` to avoid import of `torch`.

Move imports from `instructlab.eval` into function to avoid import of
`torch`.

Add a test case to test for slow and heavy imports like `torch` and
`llama_cpp`. 

Allow `import-outside-toplevel` globally.

**Issue resolved by this Pull Request:**
See: #1467
See: https://github.com/instructlab/training/pull/116

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by JamesKunstle — APPROVED** [review]
lgtm, very helpful changes!

File: src/instructlab/model/evaluate.py [hunk 1]
```diff
@@ -204,8 +208,7 @@ def qa_pairs_to_qna_to_avg_scores(qa_pairs: list[dict]) -> dict:
 )
 @click.option(
     "--benchmark",
-    type=click.Choice(list(BENCHMARK_TO_CLASS_MAP.keys())),
-    # case_sensitive=False,
+    type=click.Choice(["mmlu", "mmly_branch", "mt_branch", "mt_bench_branch"]),
```
Comment (nathan-weinberg) [inline src/instructlab/model/evaluate.py:211]:
```suggestion
    type=click.Choice(["mmlu", "mmlu_branch", "mt_branch", "mt_bench_branch"]),
```

**Review by nathan-weinberg — CHANGES_REQUESTED** [review]
Has evaluation been tested with these changes?

File: src/instructlab/model/evaluate.py (hunk 1 above)
Comment (tiran) [reply src/instructlab/model/evaluate.py:211]: Sigh ... And that's why ad-hoc string constants are bad and enums are better.

**Review by leseb — COMMENTED** [review]
How can we prevent new code additions from slowing down the CLI again? How about adding a CI test that compares the startup time before and after a new code addition? Thanks!

**Comment by tiran** [comment]
I have included a test case that will fail if `instructlab.lab` code starts depending on `torch`, `vllm`, or `llama_cpp` packages again. The test runs Python in a subprocess and blocks these modules before it tries to import `instructlab.lab`.

I don't like time-based tests, because they depend on the performance of the local machine.

**Comment by leseb** [comment]
> I have included a test case that will fail if `instructlab.lab` code starts depending on `torch`, `vllm`, or `llama_cpp` packages again. The test runs Python in a subprocess and blocks these modules before it tries to import `instructlab.lab`.
> 
> I don't like time-based tests, because they depend on the performance of the local machine.

I saw the test and it is a good guard rail but does not help understand the startup time.
I understand your concern about the time-based test. However this test will always run on the CI runner, the test will be local, and so will be the diff so it does not matter how performant it is. The CI does not have to fail, it can simply print the startup time before and after new code is being added.

**Comment by tiran** [comment]
Can you work on a timing check in a subsequent PR?

**Comment by leseb** [comment]
> Can you work on a timing check in a subsequent PR?

Will do!

**Review by leseb — APPROVED** [review]

**Comment by danmcp** [comment]
I've tested out the changes with evaluate.  Changes look good.  And much faster!

**Review by nathan-weinberg — APPROVED** [review]

**Review by danmcp — APPROVED** [review]

**Pull request merged**